### PR TITLE
Initial map setup uses env variables

### PR DESCRIPTION
**Pull request by JensWinter**
- initial location => NEXT_PUBLIC_MAP_INITIAL_LATITUDE + NEXT_PUBLIC_MAP_INITIAL_LONGITUDE
- initial zoom factor => NEXT_PUBLIC_MAP_INITIAL_ZOOM + NEXT_PUBLIC_MAP_INITIAL_ZOOM_MOBILE

Resolves https://github.com/technologiestiftung/giessdenkiez-de/issues/560

### CODE REVIEW

**Comment by ff6347** [comment]
Can you do all PRs based on and against the staging branch, please.

File: src/components/TreesMap/TreesMap.tsx [hunk 1]
```diff
@@ -122,10 +122,15 @@ if (
   `);
 }
 
+const initialLatitude = +process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE || 52.500869;
```
Comment (ff6347) [inline src/components/TreesMap/TreesMap.tsx:125]: Why the +process.env.VARNAME?

**Comment by ff6347** [comment]
> Can you do all PRs based on and against the staging branch, please.

Did that. To make the tests work the variables need to be present in .github/workflows/test.yml

File: src/components/TreesMap/TreesMap.tsx (hunk 1 above)
Comment (JensWinter) [reply src/components/TreesMap/TreesMap.tsx:125]: This gets the env var value and automatically returns it as number type. Do you prefer a different approach?

**Comment by JensWinter** [comment]
> > Can you do all PRs based on and against the staging branch, please.
> 
> Did that. To make the tests work the variables need to be present in .github/workflows/test.yml

Hm. That's strange. It works for me though. The test run states it can't find `NEXT_PUBLIC_MAP_BOUNDING_BOX`which should have always been there. It doesn't refer to the new env vars at all:
image

Can you please check if the bounding box secret is set up correctly in the Github environment?

The new variables shouldn't be a problem. They are used optionally and shouldn't raise an error in the case they are missing.

File: src/components/TreesMap/TreesMap.tsx (hunk 1 above)
Comment (ff6347) [reply src/components/TreesMap/TreesMap.tsx:125]: Nice. I did not know that this works.


```ts
describe("initialLatitude", () => {
	it("should parse string value to a number", () => {
		process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE = "2324";
		const initialLatitude =
			+process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE || 52.500869;
		expect(initialLatitude).toBe(2324);
	});

	it("should use default value when environment variable is undefined", () => {
		process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE = undefined;
		const initialLatitude =
			+process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE || 52.500869;
		expect(initialLatitude).toBe(52.500869);
	});

	it("should use default value when environment variable is null", () => {
		process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE = null;
		const initialLatitude =
			+process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE || 52.500869;
		expect(initialLatitude).toBe(52.500869);
	});

	it("should parse number value to a number", () => {
		process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE = "123";
		const initialLatitude =
			+process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE || 52.500869;
		expect(initialLatitude).toBe(123);
	});

	it("should use default value when environment variable cannot be parsed to a number", () => {
		process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE = "invalid";
		const initialLatitude =
			+process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE || 52.500869;
		expect(initialLatitude).toBe(52.500869);
	});
});
```


Only thing is the Typescript compiler throws an error.

```text
error TS18048: 'process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE' is possibly 'undefined'.
```

So something more explicit would be better.

```ts
const initialLatitude =
	!process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE ||
	isNaN(parseFloat(process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE))
		? 52.500869
		: parseFloat(process.env.NEXT_PUBLIC_MAP_INITIAL_LATITUDE);

console.log(initialLatitude);
```

**Comment by ff6347** [comment]
> > > Can you do all PRs based on and against the staging branch, please.
> > 
> > 
> > Did that. To make the tests work the variables need to be present in .github/workflows/test.yml
> 
> Hm. That's strange. It works for me though. The test run states it can't find `NEXT_PUBLIC_MAP_BOUNDING_BOX`which should have always been there. It doesn't refer to the new env vars at all: image
> 
> Can you please check if the bounding box secret is set up correctly in the Github environment?
> 
> The new variables shouldn't be a problem. They are used optionally and shouldn't raise an error in the case they are missing.

I think this is related to #563

**Comment by ff6347** [comment]
@JensWinter We moved all the env vars into the workflow and everything works fine. Can you update your branch with our staging?

**Comment by ff6347** [comment]
@dnsos @vogelino Any preferences about coding style and `tsc` happiness?

See https://github.com/technologiestiftung/giessdenkiez-de/pull/561#discussion_r1167716352

**Comment by dnsos** [comment]
> @dnsos @vogelino Any preferences about coding style and tsc happiness?

Is there still a TypeScript issue? I can see that the CI is running successfully now. Style-wise I don't see any issue with this way of converting to a number.

**Review by dnsos — APPROVED** [review]

**Comment by ff6347** [comment]
@all-contributors please add @JensWinter for code

**Pull request merged**